SConstruct

```diff
@@ -521,13 +521,13 @@ o.Add(
 o.Add(
 	"VDB_INCLUDE_PATH",
 	"The path to the OpenVDB include directory.",
-	"/usr/local/include",
+	"",
```

Comment:
This had a very "interesting" side effect for my builds, which took a while to track down. The symptom was a failure in the configure check for openvdb, with this error :

```
conftest_8.o: relocation R_X86_64_32S against symbol `_ZNKSt5ctypeIcE8do_widenEc' can not be used when making a shared object; recompile with -fPIC
```
It turned out that I had omitted to specify VDB_INCLUDE_PATH in my config, so what was `-isystem /usr/local/include` became just `-isystem `. And it just so happened that the argument after that was the all-important `-fPIC`, which was now being interpreted as an include path (`-system -fPIC`). This is totally my mistake, but it took me some time to figure out what was going in.

Can we make `append/prependSystemDirectory()` skip empty paths to avoid this pitfall? It'll be much easier to debug a "can't find header" error than a "relocation R_X86_64_32S against symbol" one.

Reply:
I'm late to the party, so please excuse if I'm getting the wrong end of the stick with any of this.

I've just done a test build with the setup used here at IE and I've run into a similar issue as you, John. Took me a little bit to hunt this down as well. 
Here at IE we do not specify `FREETYPE_INCLUDE_PATH`. I'm not sure why that's the case, but it causes our builds to lose the `-std=c++11` argument because of an `-isystem` that isn't followed by a path. The result was a whole bunch of errors regarding cpp11 keywords like `nullptr` and `override` etc.
Note that this is _with_ 1a6f3f4. I think that fix doesn't work because we don't need to check the entry in `include_list` but the path _after_ substitution instead. `"$FREETYPE_INCLUDE_PATH"` isn't `""`, but the value of that variable is (at least on our system).
In my investigation I got things to run by doing 

`include_list = [i for i in include_list if e.subst( i ) != ""]`

instead of what 1a6f3f4 proposes.

Btw., thanks for your efforts, Eric! :) It's awesome that you've taken this on. 

Just a quick note regarding review workflow: We at IE try to get into the habit of addressing John's feedback by responding with the commit hash that addresses it and then we let _him_ resolve the conversation once he's happy. I've unresolved the conversation for now so that we can get this sorted :)

Thanks!

Reply:
> I think that fix doesn't work because we don't need to check the entry in include_list but the path after substitution instead. "$FREETYPE_INCLUDE_PATH" isn't "", but the value of that variable is (at least on our system).
In my investigation I got things to run by doing
>
>`include_list = [i for i in include_list if e.subst( i ) != ""]`

Good catch Matti - your suggested solution sounds good too. Could you add that please Eric?

Reply:
Thanks for taking a look at it Matti, that does make more sense. I've resolved that with 000ab6bfdcbcdb3f1180ea73b5daac5b6d54d363

And thanks for the note about review workflow!

Reply:
Thanks Eric! Matti, could you give this one more test at your end, and merge if all seems well?

Reply:
I did, see comment below. Thanks for the fix, Eric!